include/godot_cpp/variant/variant.hpp

```diff
@@ -42,6 +42,41 @@
 
 namespace godot {
 
+class String;
+class Vector2;
+class Vector2i;
+class Rect2;
+class Rect2i;
+class Vector3;
+class Vector3i;
+class Transform2D;
+class Vector4;
+class Vector4i;
+class Plane;
+class Quaternion;
+class AABB;
+class Basis;
+class Transform3D;
+class Projection;
+class Color;
+class StringName;
+class NodePath;
+class RID;
+class Object;
+class Callable;
+class Signal;
+class Dictionary;
+class Array;
+class PackedByteArray;
+class PackedInt32Array;
+class PackedInt64Array;
+class PackedFloat32Array;
+class PackedFloat64Array;
+class PackedStringArray;
+class PackedVector2Array;
+class PackedVector3Array;
+class PackedColorArray;
```

Comment:
I'm not convinced by this change.

It's expected in Godot code that if you include `variant.h` you have all Variant definitions.

Reply:
The point of this change is to reduce the amount of things that get included & exposed, which improves the compile times.
This header gets included by a bunch generated files (e.g animation.hpp) and a couple of internal ones - so as soon as any of them get included, all built-in types are includes as-well.

As a library user, I want to decide what I really need & keep the compile times as low as possible.
I hope this convinces you :)

Reply:
Any movement on this? Maybe we can get the non-contested changes in and the variant stuff could be a separate PR?

Building this is quite slow, so any changes to improve that would be great!